Baseline Temporal Augmentation

joos2010kj opened the pull request:
Implementation of window slice

- works in progress: std/mean for Germany

joos2010kj committed:
Add crs for planetfusion_germany

joos2010kj committed:
Add window slice

ivanzvonkov commented:
Looks good, just might need some extra code to do window slice at test time

joos2010kj committed:
Merge branch 'main' into germany

Conflicts:
	helper.py

joos2010kj committed:
Add Germany mean and std

joos2010kj commented:
Ran a test after merging with main and it's working fine with germany

joos2010kj committed:
Update PlanetTransform class variables

joos2010kj committed:
Rearrange args

joos2010kj committed:
Add jitter

joos2010kj commented:
> Looks good, just might need some extra code to do window slice at test time

WIP

ivanzvonkov commented:
Can this be merged?

joos2010kj committed:
Update temp_drop in sent1 and sent2 & confirm tempaug being applied to every element in each iter

joos2010kj committed:
Update train.py params for wandb config

joos2010kj committed:
Train.py param auto-optimized

ivanzvonkov commented on src/utils/data_transform.py:
```diff
@@ -134,17 +144,25 @@ class PlanetTransform(EOTransformer):
+        elif competition == "germany":
+            PlanetTransform.per_band_mean = np.array([513.8759,  701.9154,  810.2146, 2778.312])
+            PlanetTransform.per_band_std = np.array([136.59517, 172.02454, 289.7672, 627.4169])
+
```
Remove this in favor of simple scaling

ivanzvonkov commented on submit.py:
```diff
@@ -143,11 +148,15 @@
 
 np.savez(
-    f"{name}.npz",
+    f"result/{name}.npz",
```
Saving as csv would make more sense with columns model_id, wandb_url, test accuracy, test loss

joos2010kj replied:
iirc, you already implemented changes to this in your commit, so I'll not update this part.

joos2010kj replied:
apparently it was a miscommunication, so I'll fix this

joos2010kj committed:
Final touches before baseline setup & remove norm patch from prev commit

joos2010kj committed:
Update default LR

joos2010kj commented:
To be merged after one last test after Slurm fix

ivanzvonkov committed:
Record current amount of files

ivanzvonkov committed:
Script for creating dataset stats csv

ivanzvonkov committed:
Black formatting on save

ivanzvonkov committed:
Use spatial stats by default

ivanzvonkov committed:
Allow spatial stats

ivanzvonkov committed:
Use transform directly and black formatting

ivanzvonkov committed:
Disable normalization

ivanzvonkov committed:
Use precomputed csvs for stats

ivanzvonkov committed:
Add test csv, clean up

joos2010kj committed:
Update s1g_redflag filter in files

joos2010kj committed:
Merge branch 'main' of github.com:ivanzvonkov/tum-planet-radearth-ai4food-challenge into germany

Conflicts:
	train.py

joos2010kj committed:
[WIP] Crash when python train.py --sat s1_s2 or planet_5day

ivanzvonkov committed:
Merge pull request #19 from ivanzvonkov/reproduce-germany-result

Use spatial stats for training (saved in csvs)

joos2010kj committed:
Fix S1_S2 issue

joos2010kj committed:
Fix planet_5day issue

joos2010kj committed:
[WIP] python s1_s2 run crashes

python train.py --num_epochs=40 --planet_temporal_dropout=0 --satellite=s1_s2 --window_slice=0.9 --disable_wandb

s1_s2_reader.py:
realign = (s1_timesteps != self.s1_reader.timesteps).any() or (s2_timesteps != self.s2_reader.timesteps).any()
bool.any()

joos2010kj committed:
Update gitignore

joos2010kj committed:
Merge branch 'germany' into main

Conflicts:
	.gitignore
	helper.py
	src/utils/data_transform.py
	src/utils/planet_reader.py
	train.py
	train_ta.py

joos2010kj committed:
Update planetfusion var path

joos2010kj commented on src/utils/planet_reader.py:
```diff
@@ -48,31 +52,58 @@ def __init__(
 
         :return: None
         """
```
Please take a look closely at the changes for planet_reader.py, at this region, b/c I was not so familiar with your changes but there were some merge conflicts here, so I decided to just merge yours in (code is running fine w/ planetdaily).

joos2010kj commented on src/utils/s1_s2_reader.py:
```diff
@@ -123,4 +126,23 @@ def __getitem__(self, idx):
+        return s1_s2_image_stack, s1_label, s1_s2_mask, s1_fid
+        """
+
+        return s1_s2_image_stack, s1_label, s1_s2_mask, s1_fid        
```
omitted timesteps for now

joos2010kj commented on train.py:
```diff
@@ -24,102 +24,112 @@
+# hyperparam
 arg_parser.add_argument("--batch_size", type=int, default=64)
 arg_parser.add_argument("--num_epochs", type=int, default=100)
+arg_parser.add_argument("--lr", type=float, default=0.001 if preference == "ivan" else 0.00001)
```
Since you are using germany dataset often from now on, make sure to use 0.00001 for baseline, not 0.001 for LR

joos2010kj commented on train.py:
```diff
@@ -24,102 +24,112 @@
+arg_parser.add_argument(
+    "--competition", 
+    type=str, 
+    default="south_africa" if preference == "ivan" else "germany",
```
By default, SA for you still.

joos2010kj commented on train.py:
```diff
@@ -24,102 +24,112 @@
-    "--pos", type=str, default="both_34", help="both_34, 34S_19E_258N, 34S_19E_259N, 33N_18E_242N"
+    "--pos", 
+    type=str, 
+    default="both_34" if preference == "ivan" else "33N_18E_242N", 
```
Update the location if using Germany

joos2010kj reviewed (commented):
@ivanzvonkov Please check this big PR!

ivanzvonkov commented on create_csv.py:
```diff
@@ -0,0 +1,197 @@
+import geopandas as gpd
```
Why is this an entirely new file? Isn't it already in main?

ivanzvonkov commented on data_overview.py:
```diff
@@ -0,0 +1,55 @@
+from pathlib import Path
+from tqdm import tqdm
```
ditto comment as above

ivanzvonkov commented on data_overview.txt:
```diff
@@ -0,0 +1,27 @@
+/cmlscratch/izvonkov/tum-planet-radearth-ai4food-challenge/data
```
Same comment as above

ivanzvonkov commented:
I don't understand what happened here, it seems that a lot of the changes already on the main branch are being introduced again? Do you know what is going on?
